config/config.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) OpenFaaS Project 2018. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+package config
+
+import (
+	"log"
+	"os"
+	"strings"
+	"time"
+)
+
+type Config struct {
+	GatewayURL      string
+	UpstreamTimeout time.Duration
+	Topics          []string
+	PrintResponse   bool
+	RebuildInterval time.Duration
+	Broker          string
+}
+
+func Get() Config {
+	broker := "nats"
+	if val, exists := os.LookupEnv("broker_host"); exists {
+		broker = val
+	}
+
+	topics := []string{}
+	if val, exists := os.LookupEnv("topics"); exists {
+		for _, topic := range strings.Split(val, ",") {
+			if len(topic) > 0 {
+				topics = append(topics, topic)
+			}
+		}
+	}
+	if len(topics) == 0 {
+		log.Fatal(`Provide a list of topics i.e. topics="payment_published,slack_joined"`)
+	}
+
+	gatewayURL := "http://gateway:8080"
+	if val, exists := os.LookupEnv("gateway_url"); exists {
+		gatewayURL = val
+	}
+
+	upstreamTimeout := time.Second * 30
+	rebuildInterval := time.Second * 3
+
+	if val, exists := os.LookupEnv("upstream_timeout"); exists {
+		parsedVal, err := time.ParseDuration(val)
+		if err == nil {
+			upstreamTimeout = parsedVal
+		}
+	}
+
+	if val, exists := os.LookupEnv("rebuild_interval"); exists {
+		parsedVal, err := time.ParseDuration(val)
+		if err == nil {
+			rebuildInterval = parsedVal
+		}
+	}
+
+	printResponse := false
+	if val, exists := os.LookupEnv("print_response"); exists {
+		printResponse = (val == "1" || val == "true")
+	}
+
+	return Config{
+		GatewayURL:      gatewayURL,
+		UpstreamTimeout: upstreamTimeout,
+		Topics:          topics,
+		RebuildInterval: rebuildInterval,
+		Broker:          broker,
+		PrintResponse:   printResponse,
+	}
+}
```

Comment:
How does go.mod etc work with using Go 1.9.2 in the dockerfile? Maybe dep would be better until everything else can move across?